sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/AbstractOAuth2AuthorizationCodeGrantRequestEntityConverter.java

```diff
@@ -20,25 +19,17 @@
 public abstract class AbstractOAuth2AuthorizationCodeGrantRequestEntityConverter
     extends OAuth2AuthorizationCodeGrantRequestEntityConverter {
 
+    protected AbstractOAuth2AuthorizationCodeGrantRequestEntityConverter() {
+        addHeadersConverter((request) -> getHttpHeaders());
+        addParametersConverter(this::getHttpBody);
+    }
+
     /**
      * Gets the application ID.
      *
      * @return the application ID
      */
     protected abstract String getApplicationId();
-
-    @Override
-    @SuppressWarnings("unchecked")
-    public RequestEntity<?> convert(OAuth2AuthorizationCodeGrantRequest request) {
-        addHeadersConverter(headersConverter);
-        addParametersConverter(parametersConverter);
-        return super.convert(request);
-    }
-
-    private final Converter<OAuth2AuthorizationCodeGrantRequest, HttpHeaders> headersConverter = (request) -> getHttpHeaders();
-
-    private final Converter<OAuth2AuthorizationCodeGrantRequest, MultiValueMap<String, String>> parametersConverter = this::getHttpBody;
-
     /**
      * Additional default headers information.
      * @return HttpHeaders
```

Comment:
What do you think about making this `public HttpHeaders getHttpHeaders(OAuth2AuthorizationCodeGrantRequest request) {`?

This is an abstract class, it seems like it seems like it might be nice to give implementors access to the request parameter

Reply:
Will do.